src/main/scala/no/ndla/conceptapi/service/search/PublishedConceptSearchService.scala

```diff
@@ -0,0 +1,212 @@
+
+import java.util.concurrent.Executors
+
+import cats.implicits._
+import com.sksamuel.elastic4s.http.ElasticDsl._
+import com.sksamuel.elastic4s.searches.queries.BoolQuery
+import com.typesafe.scalalogging.LazyLogging
+import no.ndla.conceptapi.ConceptApiProperties
+import no.ndla.conceptapi.integration.Elastic4sClient
+import no.ndla.conceptapi.model.api
+import no.ndla.conceptapi.model.api.{OperationNotAllowedException, ResultWindowTooLargeException, SubjectTags}
+import no.ndla.conceptapi.model.domain.{Language, SearchResult}
+import no.ndla.conceptapi.model.search.SearchSettings
+import no.ndla.conceptapi.service.ConverterService
+import no.ndla.mapping.ISO639
+
+import scala.annotation.tailrec
+import scala.concurrent.duration._
+import scala.concurrent._
+import scala.language.postfixOps
+import scala.util.{Failure, Success, Try}
+
+trait PublishedConceptSearchService {
+  this: Elastic4sClient
+    with SearchService
+    with PublishedConceptIndexService
+    with ConverterService
+    with SearchConverterService =>
+  val publishedConceptSearchService: PublishedConceptSearchService
+
+  class PublishedConceptSearchService extends LazyLogging with SearchService[api.ConceptSummary] {
+    override val searchIndex: String = ConceptApiProperties.PublishedConceptSearchIndex
+
+    override def hitToApiModel(hitString: String, language: String): api.ConceptSummary =
+      searchConverterService.hitAsConceptSummary(hitString, language)
+
+    def getTagsWithSubjects(subjectIds: List[String],
+                            language: String,
+                            fallback: Boolean): Try[List[api.SubjectTags]] = {
+      if (subjectIds.size <= 0) {
+        Failure(OperationNotAllowedException("Will not generate list of subject tags with no specified subjectIds"))
+      } else {
+        implicit val ec: ExecutionContextExecutor =
+          ExecutionContext.fromExecutor(Executors.newFixedThreadPool(subjectIds.size))
+        val searches = subjectIds.traverse(subjectId => searchSubjectIdTags(subjectId, language, fallback))
+        Await.result(searches, 1 minute).sequence.map(_.flatten)
+      }
+    }
+
+    private def searchSubjectIdTags(subjectId: String, language: String, fallback: Boolean)(
+        implicit executor: ExecutionContext): Future[Try[List[SubjectTags]]] =
+      Future {
+        val settings = SearchSettings.empty.copy(
+          subjects = Set(subjectId),
+          searchLanguage = language,
+          fallback = fallback
+        )
+
+        searchUntilNoMoreResults(settings).map(searchResults => {
+          val tagsInSubject = for {
+            searchResult <- searchResults
+            searchHits <- searchResult.results
+            matchedTags <- searchHits.tags.toSeq
+          } yield matchedTags
+
+          searchConverterService
+            .groupSubjectTagsByLanguage(subjectId, tagsInSubject)
+            .filter(tags => tags.language == language || language == Language.AllLanguages || fallback)
```

Comment:
Hmm. Nøkkelord for forklaringer er egentlig ikkje nøkkelord og brukes for strukturering av filterhierarki. Må sjekke om vi må gjøre disse språkuavhengige for forklaringer.

Reply:
Jeg lar dette være foreløpig og tar det heller i en egen oppgave om vi finner ut av vi må gjøre det. Enig?

Reply:
Enig. Vi må nok ha språkhåndtering av dette uansett.